### DISCUSSION TITLE
404 Not Found when using web workers

### DISCUSSION THREAD

**Discussion by JayJohn1869** [vercel/next.js#14051] [Help]
Hello,
I'm trying to use a web worker to offload some computationally expensive tasks away from the main thread

I have a typescript codebase and have the following code:

In `Index.tsx`
```javascript
// Also tried importing "worker-loader!../example.worker"
import ExampleWorker from "../example.worker";


export default function App() {
  const worker = useRef();

  useEffect(()=>{
    worker.current = new ExampleWorker();
  }, []);

  return (
    <button
      onClick={()=>worker.current.postMessage("runTask")}
    >Click me</button>
  );
}
```

In `example.worker.ts`:
```javascript
self.addEventListener("message", (e)=>{
  // Do some long-running operation, eg. Android Studio's indexing
})
```

It tries fetching `example.worker.js` from `/_next/[hash].worker.js` and that causes a 404 Not Found. 
The problem is, I can see the file if I visit that path through the filesystem!

I've tried fiddling with the `publicPath` and `name` options, to no avail.

Any help? Thanks!

**Comment by BrennerSpear** [comment]
same problem :(

**Reply by 3cL1p5e7** [reply]
Same
